src/sagemaker/cli/compatibility/v2/modifiers/serde.py

```diff
@@ -0,0 +1,407 @@
+
+        Returns:
+            ast.ImportFrom: a node that represents a import statement, which has
+                been modified to remove imported serializers. If nothing is
+                imported, None is returned.
+        """
+        i = 0
+        while i < len(node.names):
+            name = node.names[i].name
+            if name in OLD_CLASS_NAMES | OLD_OBJECT_NAMES:
+                node.names.pop(i)
+            else:
+                i += 1
+
+        return node if node.names else None
+
+
+class SerdeImportFromAmazonCommonRenamer(Modifier):
+    """A class to update import statements starting with ``from sagemaker.amazon.common``."""
+
+    def node_should_be_modified(self, node):
+        """Checks if the import statement imports a SerDe from the
+        ``sagemaker.amazon.common`` module.
+
+        This checks for:
+        - ``sagemaker.amazon.common.numpy_to_record_serializer``
+        - ``sagemaker.amazon.common.record_deserializer``
+
+        Args:
+            node (ast.ImportFrom): a node that represents a ``from ... import ... `` statement.
+                For more, see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            bool: True if and only if the ``ast.ImportFrom`` imports a SerDe from
+                the ``sagemaker.amazon.common`` module.
+        """
+        return node.module == "sagemaker.amazon.common" and any(
+            [alias.name in OLD_AMAZON_CLASS_NAMES for alias in node.names]
+        )
+
+    def modify_node(self, node):
+        """Upgrades the ``numpy_to_record_serializer`` and ``record_deserializer``
+        imports, as applicable.
+
+        This upgrades the classes to:
+        - ``sagemaker.amazon.common.RecordSerializer``
+        - ``sagemaker.amazon.common.RecordDeserializer``
+
+        Args:
+            node (ast.ImportFrom): a node that represents a ``from ... import ... `` statement.
+                For more, see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            ast.ImportFrom: a node that represents a import statement, which has
+                been modified to import the upgraded class name.
+        """
+        for alias in node.names:
+            if alias.name in OLD_AMAZON_CLASS_NAMES:
+                alias.name = OLD_CLASS_NAME_TO_NEW_CLASS_NAME[alias.name]
+        return node
+
+
+class _ImportInserter(Modifier):
+    """A class to insert import statements into the Python module."""
+
+    def __init__(self, class_names, import_node):
+        """Initialize the ``class_names`` and ``import_node`` attributes.
+
+        Args:
+            class_names (set): If any of these class names are referenced in the
+                module, then ``import_node`` is inserted.
+            import_node (ast.ImportFrom): The AST node to insert.
+        """
+        self.class_names = class_names
+        self.import_node = import_node
```

Comment:
I'm not sure if one way provides any benefit over the other, but you could also make these abstract property methods

Reply:
I thought about it, and I can't really decide which is better, so I'm leaving it as-is.